docs/reference/glossary.rst

```diff
@@ -0,0 +1,97 @@
+.. _glossary:
+
+========
+Glossary
```

Comment:
Maybe some things should link to e.g. http://doc-dev.clusterhq.com/concepts/architecture.html for more information.